### PR TITLE
feat: Add support for parsing the syntax of MySQL UNIQUE KEY.

### LINKED ISSUE #961: Error: SchemaParse(ParserError("Expected a list of columns in parentheses, found: KEY"))
The SQL statement I want to parse is: 
```sql
CREATE TABLE `funds_sale` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT COMMENT '',
  PRIMARY KEY (`id`, `cid`),
  UNIQUE KEY `uk_bill_str` (`cid`, `bill_str`),
  KEY `idx_biz_str` (`cid`, `biz_str`)
  ) ENGINE = InnoDB AUTO_INCREMENT = 878936 DEFAULT CHARSET = utf8mb4 COMMENT = ''

```
 but I encountered an error.
```rust
Error: SchemaParse(ParserError("Expected a list of columns in parentheses, found: KEY"))
```

### PR DESCRIPTION
**Pull request by artorias1024**
Fixes #961

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -3894,9 +3894,15 @@ impl<'a> Parser<'a> {
         match next_token.token {
             Token::Word(w) if w.keyword == Keyword::PRIMARY || w.keyword == Keyword::UNIQUE => {
                 let is_primary = w.keyword == Keyword::PRIMARY;
-                if is_primary {
+                let name = if is_primary {
```
Comment (alamb) [inline src/parser/mod.rs:3897]: I wonder if this code could be simpler if it didn't check for `is_primary` and just always did the `self.maybe_parse` clause?

**Review by alamb — APPROVED** [review]
Thank you for the contribution @artorias1024  - I think this code could be simpler (I left a suggestion) but otherwise it looks good to me. Let me know what you think

I apologize for the delay in reviewing -- I was away last week

File: src/parser/mod.rs (hunk 1 above)
Comment (artorias1024) [reply src/parser/mod.rs:3897]: Sorry, I have no idea how to optimize this code...      T T

**Comment by alamb** [comment]
Thanks @artorias1024  -- I took the liberty of merging `main` into this branch to get a fix for a logical merge conflict and I also added [137584c](https://github.com/sqlparser-rs/sqlparser-rs/pull/962/commits/137584cc7aafb35cd02125b0b484f2a6e968a685) to simplify the logic. What do you think?

**Comment by artorias1024** [comment]
> Thanks @artorias1024 -- I took the liberty of merging `main` into this branch to get a fix for a logical merge conflict and I also added [137584c](https://github.com/sqlparser-rs/sqlparser-rs/pull/962/commits/137584cc7aafb35cd02125b0b484f2a6e968a685) to simplify the logic. What do you think?

Thanks

**Comment by alamb** [comment]
Thanks again @artorias1024

**Pull request merged**